apps/server/src/shared/domain/service/lesson-copy.service.ts

```diff
@@ -8,21 +9,32 @@ export type LessonCopyParams = {
 	user: User;
 };
 
-export type LessonCopyResponse = {
-	// copy: Lesson;
-	status: CopyStatus;
-};
-
 @Injectable()
 export class LessonCopyService {
-	copyLesson(params: LessonCopyParams): LessonCopyResponse {
-		const result = {
-			status: {
-				title: params.originalLesson.name,
-				type: CopyElementType.LESSON,
-				status: CopyStatusEnum.NOT_IMPLEMENTED,
-			},
+	constructor(private readonly nameCopyService: NameCopyService) {}
+
+	copyLesson(params: LessonCopyParams): CopyStatus {
+		const copy = new Lesson({
+			course: params.destinationCourse,
+			hidden: params.originalLesson.hidden,
+			name: this.nameCopyService.deriveCopyName(params.originalLesson.name),
+			position: params.originalLesson.position,
```

Comment:
also unclear what should be the behaviour of position

Reply:
maybe this comes from the legacy functionality? otherwise it might be that its not even used anymore as we have a special sorting in the room view